server/src/main/java/org/candlepin/resource/RoleResource.java

```diff
@@ -99,150 +164,124 @@ public Role createRole(@ApiParam(name = "role", required = true) Role role) {
         " permissions.", value = "updateRole")
     @ApiResponses({ @ApiResponse(code = 404, message = "") })
     @PUT
-    @Path("{role_id}")
+    @Path("{role_name}")
```

Comment:
Are we free to change the API here? (from role_id to role_name)

Reply:
Yup, no one uses it (yet), so we're free to change it to be more in-line with the user APIs and other such objects which pass through the adapters (and, thus, can't be expected to know about our internal DB IDs).